DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UnsupportedViewRecorder.swift

```diff
@@ -0,0 +1,74 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import UIKit
+import WebKit
+import SwiftUI
+
+internal struct UnsupportedViewRecorder: NodeRecorder {
+    // swiftlint:disable opening_brace
+    private let unsupportedViewsPredicates: [(UIView) -> Bool] = [
+        { $0 is UIProgressView },
+        { $0 is UIWebView },
+        { $0 is WKWebView },
+        { $0 is UIActivityIndicatorView },
+        { String(reflecting: type(of: $0)).contains("SwiftUI") }
+    ]
+    // swiftlint:enable opening_brace
+
+    func semantics(of view: UIView, with attributes: ViewAttributes, in context: ViewTreeRecordingContext) -> NodeSemantics? {
+        guard unsupportedViewsPredicates.contains(where: { $0(view) }) else {
+            return nil
+        }
+        guard attributes.isVisible else {
+            return InvisibleElement(subtreeStrategy: .ignore)
+        }
+        let builder = UnsupportedViewWireframesBuilder(
+            wireframeRect: view.frame,
+            wireframeID: context.ids.nodeID(for: view),
+            unsupportedClassName: String(reflecting: type(of: view)).extractGenericClassNameIfNeeded(),
+            attributes: attributes
+        )
+        let node = Node(viewAttributes: attributes, wireframesBuilder: builder)
+        return SpecificElement(subtreeStrategy: .ignore, nodes: [node])
+    }
+}
+
+internal struct UnsupportedViewWireframesBuilder: NodeWireframesBuilder {
+    let wireframeRect: CGRect
+
+    let wireframeID: WireframeID
+    let unsupportedClassName: String
+    let attributes: ViewAttributes
+
+    func buildWireframes(with builder: WireframesBuilder) -> [SRWireframe] {
+        return [
+            builder.createTextWireframe(
+                id: wireframeID,
+                frame: attributes.frame,
+                text: unsupportedClassName,
+                textFrame: attributes.frame,
+                textAlignment: .init(horizontal: .center, vertical: .center),
+                textColor: UIColor.red.cgColor,
+                borderColor: UIColor.lightGray.cgColor,
+                borderWidth: 1,
+                backgroundColor: UIColor(white: 0, alpha: 0.05).cgColor,
+                cornerRadius: 4
+            )
+        ]
+    }
+}
+
+fileprivate extension String {
+    func extractGenericClassNameIfNeeded() -> String {
+        let regex = try? NSRegularExpression(pattern: "<([^>]*)>[^<>]*$", options: [])
+        if let match = regex?.firstMatch(in: self, range: NSRange(startIndex..., in: self)),
+            let range = Range(match.range(at: 1), in: self) {
+            return String(self[range])
+        }
+        return self
+    }
```

Comment:
Running in on the main thread (from `semantics(of:with:in:)` sounds bad idea. It makes the call 2 times slower (`~0.005ms` without → `~0.012ms` with regex). Given that we want to be performance-first and this logic is fairly easy to move to background thread (to `buildWireframes(with:)`), we should avoid spending time on the main queue if possible.

On top of that, if we want to enter regexps world, we need a fair test coverage for it (not only happy path).

On top of on top 🙂 - what does it do exactly? Is it critical 🤔? It needs some code comments (the general price for most write-only regexps 😝), otherwise it's too ambiguous.

Reply:
Yeah - I was on a fence with this. It's removes `SwiftUI._UIHostingView<>` from the class name and keeps what is between <>.

I'll remove it and we can reconsider it at later stage.